### PR TITLE
Add nspr/4.24 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **nspr/4.24**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries For macos, the CI did not warn that both `.a` and `.dylib` files were packaged.

File: recipes/nspr/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    description = "Netscape Portable Runtime (NSPR) provides a platform-neutral API for system level and libc-like functions."
+    topics = ("conan", "nspr", "libc")
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "arch", "build_type"
+    license = "MPL-2.0"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_mozilla": [True, False],
+        "win32_target": ["winnt", "win95"]
```
Comment (SSE4) [inline recipes/nspr/all/conanfile.py]: I am not sure if it's still useful to provide Windows 95 support, but if it's easy, I am okay with that

File: recipes/nspr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    def _library_names(self):
+        return ["plds4", "plc4", "nspr4"]
+
+    def package_info(self):
+        libs = self._library_names
+        if self.settings.os == "Windows" and not self.options.shared:
+            libs = list("{}_s".format(l) for l in libs)
+        self.cpp_info.libs = libs
+        if self.settings.arch == "x86":
+            self.cpp_info.defines.append("_M_IX86")
```
Comment (SSE4) [inline recipes/nspr/all/conanfile.py]: is this really needed? seems to be automatically defined by MSVC compiler:
https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019

File: recipes/nspr/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.a".format(lib)))
+                else:
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.dylib".format(lib)))
+
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.build_type == "Debug":
+                for lib in self._library_names:
+                    os.remove(os.path.join(self.package_folder, "lib", "{}.pdb".format(lib)))
+
+        if not self.options.shared or self.settings.os == "Windows":
```
Comment (SSE4) [inline recipes/nspr/all/conanfile.py]: do we need chmod for Windows?

File: recipes/nspr/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/nspr/all/conanfile.py]: These definitions are missing on mingw.
I've changed the recipe to conditionally add them.

File: recipes/nspr/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/nspr/all/conanfile.py]: I've not tested it as I have no appropriate toolchain, but it was just a matter of a configuration argument.
So maybe we make somebody happy with it...

File: recipes/nspr/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/nspr/all/conanfile.py]: Not needed per se, I just noticed these access permissions while using msys2 during development of this recipe.

**Comment by madebr** [comment]
@danimtb Final review please :relaxed:

**Pull request merged**